pkg/utils/utils.go

```diff
@@ -66,3 +67,18 @@ func GetTrimmedGroupName(group string) string {
 	}
 	return group
 }
+
+// GetStorageClassNameForPVC - Get the storageClass name from the PVC spec
+func GetStorageClassNameForPVC(pvc *v1.PersistentVolumeClaim) (string, error) {
+	var scName string
+	if pvc.Spec.StorageClassName != nil && len(*pvc.Spec.StorageClassName) > 0 {
+		scName = *pvc.Spec.StorageClassName
+	} else {
+		scName = pvc.Annotations[v1.BetaStorageClassAnnotation]
+	}
+
+	if len(scName) == 0 {
+		return "", fmt.Errorf("PVC: %s does not have a storage class", pvc.Name)
```

Comment:
Lets return a well defined error like ErrStorageClassNotDefined and check this error in the callers and not error out there. There is a case where you can manually bind a PV to a PVC (w/o a storage class). This will ensure backup/restores will work for such volumes as well.

Reply:
Here we are return only one type of error. Do we till want to have ErrStorageClassNotDefined error code definition?